jupyterhub/jupyterhub/base/jupyterhub-dc.yaml

```diff
@@ -32,26 +32,27 @@ spec:
           oc get pod -l "deploymentconfig=jupyterhub" -o jsonpath='{.items[0].metadata.labels.deployment}';
         };
 
-        function generate_payload() {
-          echo '{"kind":"Scale","apiVersion":"autoscaling/v1","metadata":{"name":"'$1'","namespace":"'${OPENSHIFT_DEPLOYMENT_NAMESPACE}'"},"spec":{"replicas": '$2'}}'
-        };
 
-        function post_scale() {
-          curl -kL -XPUT -H "Content-Type: application/json" -H "Authorization: Bearer $TOKEN" https://openshift.default.svc.cluster.local/api/v1/namespaces/${OPENSHIFT_DEPLOYMENT_NAMESPACE}/replicationcontrollers/$1/scale -d @$2;
+        function scale_replicaset() {
+          oc scale rc $1 --replicas=$2;
         };
 
+        function delete_pods() {
+          oc delete pod --force -l deployment==${OPENSHIFT_DEPLOYMENT_NAME}
+        } 
+
         function get_replicas() {
           oc get rc ${OPENSHIFT_DEPLOYMENT_NAME} -o jsonpath='{.metadata.annotations.kubectl\.kubernetes\.io\/desired-replicas}'
         }
 
-        export TOKEN=$(cat /run/secrets/kubernetes.io/serviceaccount/token);
         export OLD_RC=$(get_latest);
 
-        echo $(generate_payload ${OLD_RC} 0) > /tmp/old.yaml;
-        echo $(generate_payload ${OPENSHIFT_DEPLOYMENT_NAME} $(get_replicas)) > /tmp/new.yaml;
+        if [ -n "${OLD_RC}" ]; then
+          delete_pods
```

Comment:
should this be reversed? In this order, I think deleting the pods would just spin up new replacements, but then scaling the RS down will kill the brand new pods just created?